src/OrbitGl/GpuTrack.cpp

```diff
@@ -54,325 +46,232 @@ std::string MapGpuTimelineToTrackLabel(std::string_view timeline) {
+    case TimerInfo::kGpuDebugMarker:
+      return marker_track_->GetRight(textbox);
+    default:
+      UNREACHABLE();
+  }
 }
 
-std::string GpuTrack::GetCommandBufferTooltip(
-    const orbit_client_protos::TimerInfo& timer_info) const {
-  return absl::StrFormat(
-      "<b>Command Buffer Execution</b><br/>"
-      "<i>At `vkBeginCommandBuffer` and `vkEndCommandBuffer` `vkCmdWriteTimestamp`s have been "
-      "inserted. The GPU timestamps get aligned with the corresponding hardware execution of the "
-      "submission.</i>"
-      "<br/>"
-      "<br/>"
-      "<b>Submitted from process:</b> %s [%d]<br/>"
-      "<b>Submitted from thread:</b> %s [%d]<br/>"
-      "<b>Time:</b> %s",
-      capture_data_->GetThreadName(timer_info.process_id()), timer_info.process_id(),
-      app_->GetCaptureData().GetThreadName(timer_info.thread_id()), timer_info.thread_id(),
-      GetPrettyTime(TicksToDuration(timer_info.start(), timer_info.end())).c_str());
+const TextBox* GpuTrack::GetUp(const TextBox* textbox) const {
+  const TimerInfo& timer_info = textbox->GetTimerInfo();
+  switch (timer_info.type()) {
+    case TimerInfo::kGpuActivity:
+      [[fallthrough]];
+    case TimerInfo::kGpuCommandBuffer:
+      return submission_track_->GetUp(textbox);
+    case TimerInfo::kGpuDebugMarker:
+      return marker_track_->GetUp(textbox);
+    default:
+      UNREACHABLE();
+  }
 }
 
-std::string GpuTrack::GetDebugMarkerTooltip(
-    const orbit_client_protos::TimerInfo& timer_info) const {
-  std::string marker_text = string_manager_->Get(timer_info.user_data_key()).value_or("");
-  return absl::StrFormat(
-      "<b>Vulkan Debug Marker</b><br/>"
-      "<i>At the marker's begin and end `vkCmdWriteTimestamp`s have been "
-      "inserted. The GPU timestamps get aligned with the corresponding hardware execution of the "
-      "submission.</i>"
-      "<br/>"
-      "<br/>"
-      "<b>Marker text:</b> %s<br/>"
-      "<b>Submitted from process:</b> %s [%d]<br/>"
-      "<b>Submitted from thread:</b> %s [%d]<br/>"
-      "<b>Time:</b> %s",
-      marker_text, capture_data_->GetThreadName(timer_info.process_id()), timer_info.process_id(),
-      app_->GetCaptureData().GetThreadName(timer_info.thread_id()), timer_info.thread_id(),
-      GetPrettyTime(TicksToDuration(timer_info.start(), timer_info.end())).c_str());
+const TextBox* GpuTrack::GetDown(const TextBox* textbox) const {
+  const TimerInfo& timer_info = textbox->GetTimerInfo();
+  switch (timer_info.type()) {
+    case TimerInfo::kGpuActivity:
+      [[fallthrough]];
+    case TimerInfo::kGpuCommandBuffer:
+      return submission_track_->GetDown(textbox);
+    case TimerInfo::kGpuDebugMarker:
+      return marker_track_->GetDown(textbox);
+    default:
+      UNREACHABLE();
+  }
+}
+void GpuTrack::OnCollapseToggle(TriangleToggle::State state) {
+  Track::OnCollapseToggle(state);
+  if (state == TriangleToggle::State::kCollapsed) {
+    submission_track_->Collapse();
```

Comment:
Only the submission track or also the marker track?

Reply:
Exactly. This is somewhat of a hack. When the complete GpuTrack is collapsed, it looks exactly the same way as it would currently look if you collapse a submission track. In order to do so, we collapse the subtrack, such that the timers get collapsed.

Reply:
If this is somewhat of a hack, you should add a comment explaining why we are doing what we are doing here.

Reply:
Done.